tests/lib/IframeApi.spec.js

```diff
@@ -0,0 +1,96 @@
+describe('IframeApi', () => {
+    const iframeApi = new IFrameApi();
+
+    beforeEach(() => {
+        spyOn(CookieJar, 'eat').and.callThrough();
+        spyOn(AccountStore.instance, 'list').and.callThrough();
+        spyOn(KeyStore.instance, 'list').and.callThrough();
+        spyOn(KeyStore.instance, 'doMigrateAccountsToKeys').and.callThrough();
+        spyOnProperty(document, 'cookie', 'get').and.returnValue(Dummy.cookie);
+    });
+
+    it('can list deprecated accounts from cookies on iOS.', async () => {
+        spyOn(BrowserDetection, 'isIos').and.returnValue(true);
+
+        const listedAccounts = await iframeApi.list(true);
+
+        expect(CookieJar.eat).toHaveBeenCalledWith(true);
+        expect(AccountStore.instance.list).not.toHaveBeenCalled();
+        expect(KeyStore.instance.list).not.toHaveBeenCalled();
+        expect(listedAccounts).toEqual(Dummy.deprecatedAccountInfo);
+    });
+
+    it('can list key info from cookies on iOS', async () => {
+        spyOn(BrowserDetection, 'isIos').and.returnValue(true);
+
+        const listedKeys = await iframeApi.list();
+
+        expect(CookieJar.eat).toHaveBeenCalledWith(undefined);
+        expect(AccountStore.instance.list).not.toHaveBeenCalled();
+        expect(KeyStore.instance.list).not.toHaveBeenCalled();
+        expect(listedKeys).toEqual(Dummy.keyInfo);
```

Comment:
doesn't this only work when the cookie is set before? What happens when this method is called for the very first time?

Reply:
If no cookie present, list() should return an empty array.

Reply:
yes, it is implemented like that. I'll add a test for that in an extra PR